scripts/pre_commit_linter.py

```diff
@@ -580,6 +580,94 @@ def _check_bad_pattern_in_file(filename, content, pattern):
     return False
 
 
+def _check_bad_pattern_in_comments_docstrings(filename, content):
+    """Detects whether comments and docstrings end with a period.
+
+    Args:
+        filename: str. Name of the file.
+        content: str. Contents of the file.
+
+    Returns:
+        bool. True if there is bad pattern else false.
+    """
+    regexp_comment = r"#[^\n]*[^.,\n]\n( |\t)*(#(( |\t)*[A-Z])|[^#\s])"
+    message_comment = "There should be a period after comment end."
```

Comment:
Convert to single quote string, see the [style guide on strings](https://google.github.io/styleguide/pyguide.html?showone=Strings#Strings).

Reply:
`There should be a period at the end of the comment.`